plugins/woocommerce/client/blocks/assets/js/blocks/mini-cart/mini-cart-contents/edit.tsx

```diff
@@ -59,58 +60,16 @@ const Edit = ( {
 		defaultTemplate,
 	} );
 
-	/**
-	 * This is a workaround for the Site Editor to set the correct
-	 * background color of the Mini-Cart Contents block base on
-	 * the main background color set by the theme.
-	 */
-	useEffect( () => {
-		const canvasEl = document.querySelector(
-			'.edit-site-visual-editor__editor-canvas'
-		);
-		if ( ! ( canvasEl instanceof HTMLIFrameElement ) ) {
-			return;
-		}
-		const canvas =
-			canvasEl.contentDocument || canvasEl.contentWindow?.document;
-		if ( ! canvas ) {
-			return;
-		}
-		if ( canvas.getElementById( 'mini-cart-contents-background-color' ) ) {
-			return;
-		}
-		const styles = canvas.querySelectorAll( 'style' );
-		const [ cssRule ] = Array.from( styles )
-			.map( ( style ) => Array.from( style.sheet?.cssRules || [] ) )
-			.flatMap( ( style ) => style )
-			.filter( Boolean )
-			.filter(
-				( rule ) =>
-					rule.selectorText === '.editor-styles-wrapper' &&
-					rule.style.backgroundColor
-			);
-		if ( ! cssRule ) {
-			return;
-		}
-		const backgroundColor = cssRule.style.backgroundColor;
-		if ( ! backgroundColor ) {
-			return;
-		}
-		const style = document.createElement( 'style' );
-		style.id = 'mini-cart-contents-background-color';
-		style.appendChild(
-			document.createTextNode(
-				`:where(.wp-block-woocommerce-mini-cart-contents) {
-				background-color: ${ backgroundColor };
-			}`
-			)
-		);
-		const body = canvas.querySelector( '.editor-styles-wrapper' );
-		if ( ! body ) {
-			return;
-		}
-		body.appendChild( style );
-	}, [] );
+	// Apply the Mini-Cart Contents block base styles based on Site Editor's background and text colors.
+	useThemeColors(
+		'mini-cart-content',
```

Comment:
Small nitpick to match the block name:

```suggestion
		'mini-cart-contents',
```

Reply:
Addressed this and the linting error.

Reply:
I see a few tests are failing, and I believe they are unrelated to this PR. It's also possible that they might get resolved with this [PR](https://github.com/woocommerce/woocommerce/pull/58359), so I’ve rebased the branch.